# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1375914462 14400
#      Wed Aug 07 18:27:42 2013 -0400
# Node ID d23fcc227fa9925d502f357c4ccda6af91dc808a
# Parent  fe7cb5bb4388a00a5efcedd75666e48df15ec4e7
Account for uimenu height when determining the axes currentpoint.

* libinterp/dldfcn/__init_fltk__.cc (set_axes_currentpoint): When present,
offset the axes currentpoint by the unmenu height.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -1107,33 +1107,35 @@ private:
   void set_currentpoint (int px, int py)
   {
     if (!fp.is_beingdeleted ())
       {
         Matrix pos (1,2,0);
         pos(0) = px;
         pos(1) = h () - menu_h - py;
         if (! uimenu->is_visible ())
-          pos(1) = pos(1) +  menu_h;
+          pos(1) = pos(1) + menu_h;
         fp.set_currentpoint (pos);
         graphics_object robj = gh_manager::get_object (fp.get_parent ());
         root_figure::properties& rp =
           dynamic_cast<root_figure::properties&> (robj.get_properties ());
         rp.set_currentfigure (fp.get___myhandle__ ().value ());
       }
   }
 
   void set_axes_currentpoint (graphics_object ax, int px, int py)
   {
     if (ax.valid_object ())
       {
         axes::properties& ap =
           dynamic_cast<axes::properties&> (ax.get_properties ());
 
         double xx, yy;
+        if (uimenu->is_visible ())
+          py = py - menu_h;
         pixel2pos (ax, px, py, xx, yy);
 
         Matrix pos (2,3,0);
         pos(0,0) = xx;
         pos(1,0) = yy;
         pos(0,1) = xx;
         pos(1,1) = yy;
 
